src/sqlfluff/rules/L050.py

```diff
@@ -0,0 +1,79 @@
+"""Implementation of Rule L050."""
+from typing import Optional
+
+from sqlfluff.core.rules.base import BaseRule, LintResult, LintFix, RuleContext
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L050(BaseRule):
+    """Files must not begin with newlines or whitespace.
+
+    | **Anti-pattern**
+    | The content in file begins with newlines or whitespace, the ^ represents the beginning of file.
+
+    .. code-block:: sql
+       :force:
+
+        ^
+
+        SELECT
+            a
+        FROM foo
+
+        -- Beginning on an indented line is also forbidden, the • represents space.
```

Comment:
Avoid this unnecessarily overflowing to two lines:

```suggestion
        -- Beginning on an indented line is also forbidden,
        -- (the • represents space).
```